spec/lib/insights/api/common/request_spec.rb

```diff
@@ -93,6 +105,68 @@
     end
   end
 
+
+  context "with a good system request" do
+    let(:system_request) do
+      {
+        :headers      => {
+          'x-rh-identity'            => encoded_system_hash,
+          'x-rh-insights-request-id' => "01234567-89ab-cdef-0123-456789abcde",
+        },
+        :original_url => 'https://example.com'
+      }
+    end
+
+    around do |example|
+      described_class.with_request(system_request) do |instance|
+        @instance = instance
+        example.call
+      end
+    end
+
+    it "#original_url" do
+      expect(@instance.original_url).to eq "https://example.com"
+    end
+
+    describe "#headers" do
+      it "return a headers object" do
+        expect(@instance.headers).to be_a(ActionDispatch::Http::Headers)
+      end
+
+      it "allows case-insensitive lookup" do
+        expect(@instance.headers["X-Rh-Identity"]).to eq encoded_system_hash
+      end
+    end
+
+    it "#user" do
+      expect(@instance.user).to be_a(Insights::API::Common::User)
+    end
+
+    it "#tenant" do
+      expect(@instance.tenant).to eq(default_account_number)
+    end
+
+    it "#system" do
+      expect(@instance.system).to be_a(Insights::API::Common::System)
+    end
+
+    it "#auth_type" do
+      expect(@instance.auth_type).to eq("cert-auth")
+    end
+
+    it "#user" do
```

Comment:
Duplicate of https://github.com/RedHatInsights/insights-api-common-rails/pull/162/files#diff-b602271974e4016385b4ea6a117dd8c4R141

Reply:
:+1: deleted